core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/salt/update/SaltUpdateActions.java

```diff
@@ -116,16 +135,48 @@ protected Selectable createRequest(ClusterViewContext context) {
     @Bean(name = "SALT_UPDATE_FAILED_STATE")
     public Action<?, ?> saltUpdateFailedAction() {
         return new AbstractStackFailureAction<ClusterCreationState, ClusterCreationEvent>() {
+
+            @Inject
+            private StackDtoService stackDtoService;
+
             @Override
             protected void doExecute(StackFailureContext context, StackFailureEvent payload, Map<Object, Object> variables) {
-                saltUpdateService.handleClusterCreationFailure(context.getStack(), payload.getException());
+                Boolean checkIfBackupDatalakeDatabaseFlowChain = isBackupDatalakeDatabaseFlowChain(payload.getResourceId());
+                saltUpdateService.handleClusterCreationFailure(context.getStack(), payload.getException(), context.getStackId(),
+                        checkIfBackupDatalakeDatabaseFlowChain);
                 sendEvent(context);
             }
 
+            @Override
+            protected StackFailureContext createFlowContext(FlowParameters flowParameters,
+                StateContext<ClusterCreationState, ClusterCreationEvent> stateContext, StackFailureEvent payload) {
+                if (isBackupDatalakeDatabaseFlowChain(payload.getResourceId())) {
+                    StackView stack = stackDtoService.getStackViewById(payload.getResourceId());
+                    MDCBuilder.buildMdcContext(stack);
+                    Map<Object, Object> variables = stateContext.getExtendedState().getVariables();
+                    ProvisionType provisionType = (ProvisionType) variables.getOrDefault(PROVISION_TYPE, ProvisionType.REGULAR);
+                    return new StackFailureContext(flowParameters, stack, stack.getId(), provisionType);
+                } else {
+                    return (super.createFlowContext(flowParameters, stateContext, payload));
+                }
+            }
+
             @Override
             protected Selectable createRequest(StackFailureContext context) {
                 return new StackEvent(SaltUpdateEvent.SALT_UPDATE_FAILURE_HANDLED_EVENT.event(), context.getStackId());
             }
+
+            @VisibleForTesting
```

Comment:
please do not use this annotation, especially for public methods, you can test them without it, you should be able to test even private methods through private methods without this

Reply:
@horadla23 Thank you! I removed this annotation and set the method to be private. I will discuss with our team about how we want to address this PR regarding your other comments. TY!